sdk/identity/azure-identity/inc/azure/identity/azure_cli_credential.hpp

```diff
@@ -85,6 +85,12 @@ namespace Azure { namespace Identity {
         Core::Credentials::TokenRequestContext const& tokenRequestContext,
         Core::Context const& context) const override;
 
+    /**
+     * @brief Gets the name of the credential.
+     *
+     */
+    std::string GetCredentialName() const override;
```

Comment:
Yes - it is easier contract to implement, if we are going with the virtual method. Customer credentials can just do `return "My Credential"`. If we have a reference return type, it imposes that the class should store that string somewhere. Either is not a big deal, but I'd prefer the by-value version, especially this is not a critical path. In our code, you can see, I did implement it to utilize copy-on-write string optimization, if it is available.

BTW, if we wanted byref return, there was other design that I was considering, which is to have string parameter in the `TokenCredential` constructor + non-virtual `GetCredentialName()` - what do you think about that? I think that the current design is easier to use, that's why I didn't go with the constructor argument (I spend some time implementing it but finally abandoned it in favor of virtual method).

Reply:
I've made it non-virtual, and now it returns string const reference. This was also because we now call `GetCredentialName()` in constructors, and virtual functions do not work as virtual when called from constructors.